codepost/models/courses.py

```diff
@@ -64,18 +67,18 @@ def list_available(self, name=None, period=None):
 
         if ret.status_code == 200:
             # Returns a list of courses
-            course_list = list(map(lambda kws: _class_type(**kws), ret.json))
+            course_iter = map(lambda kws: _class_type(**kws), ret.json)
 
             # Optionally filter according to the `name` parameter
             if name:
-                course_list = filter(lambda c: c.name == name, course_list)
+                course_iter = filter(lambda c: c.name == name, course_iter)
 
             # Optionally filter according to the `period` parameter
             if period:
-                course_list = filter(lambda c: c.period == period, course_list)
+                course_iter = filter(lambda c: c.period == period, course_iter)
 
-            return course_list
+            return course_iter
```

Comment:
@jlumbroso. Here's another design decision related to uniqueness. 

By specifying both a `name` and a `period` here, the client restricts herself to 0 or 1 results. Yet we still return a list in this situation. Is this a good case to return `None` or object in this case? 

I think not because:
* these functions are explicitly designed to return lists
* switching between return types is therefore likely to be confusing

But thought I'd flag in case you (unexpectedly) had a different opinion.

Reply:
@jamesaevans Yes, we are exactly in agreement. This is a good example of a situation where I don't think it's appropriate to return an object *or* a list depending on the situation.